src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+        for (var {|Rename:i|} = 0; i < array.Length; i++) /* test */
+        {
+            var a = array[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment6()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array) 
+            /* test */ Console.WriteLine(a); 
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < array.Length; i++)
+        {
+            var a = array[i];
+            /* test */
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment7()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        var {|Rename:list|} = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < list.Length; i++)
+        {
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a /* test */ in array) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported2()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (/* test */ var a in array) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported3()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array /* test */) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CollectionStatement()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        foreach[||](var a in new int[] { 1, 3, 4 })
```

Comment:
have test where 'foreach' is an embedded statement.  i.e. directly parented by an if statement.

Reply:
that's good one. for now, I probably won't offer refactoring in that case.

Reply:
why not?

Reply:
Note: if the collection is written into, i still offer ConvertForToForEach, i just put a warning in the code.

Reply:
because it needs to convert destination to block based on context. expression body, embedded statement and etc. what this is doing is basically what introduce variable refactoring is doing. currently that service is too tightly coupled to refactoring. we probably support it by refactoring that service and use that service here to extract out to local and work on that code.

Reply:
tracking in this issue - https://github.com/dotnet/roslyn/issues/25667